web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/GetUserResourceLinksImpl.java

```diff
@@ -45,50 +44,67 @@ private ArrayList<ResourceLinkDTO> getAvailableFunctionsForUser(WebCertUser user
 
         if (hasAccessToSearchCreatePage(user)) {
             resourceLinks.add(
-                    ResourceLinkDTO.create(
-                            ResourceLinkTypeDTO.ACCESS_SEARCH_CREATE_PAGE,
-                            "Sök / skriv intyg",
-                            "",
-                            true
-                    )
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.ACCESS_SEARCH_CREATE_PAGE,
+                    "Sök / skriv intyg",
+                    "",
+                    true
+                )
             );
         }
 
         if (hasAccessToDraftList(user)) {
             resourceLinks.add(
-                    ResourceLinkDTO.create(
-                            ResourceLinkTypeDTO.ACCESS_DRAFT_LIST,
-                            "Ej signerade utkast",
-                            "",
-                            true
-                    )
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.ACCESS_DRAFT_LIST,
+                    "Ej signerade utkast",
+                    "",
+                    true
+                )
             );
         }
 
         if (hasAccessToSignedCertificatesList(user)) {
             resourceLinks.add(
-                    ResourceLinkDTO.create(
-                            ResourceLinkTypeDTO.ACCESS_SIGNED_CERTIFICATES_LIST,
-                            "Signerade intyg",
-                            "",
-                            true
-                    )
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.ACCESS_SIGNED_CERTIFICATES_LIST,
+                    "Signerade intyg",
+                    "",
+                    true
+                )
             );
         }
 
         if (isLogOutAvailable(user)) {
             resourceLinks.add(
-                    ResourceLinkDTO.create(
-                            ResourceLinkTypeDTO.LOG_OUT,
-                            "Logga ut",
-                            "",
-                            true
-                    )
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.LOG_OUT,
+                    "Logga ut",
+                    "",
+                    true
+                )
+            );
+        }
+
+        if (isChooseUnitAvailable(user)) {
+            resourceLinks.add(
+                ResourceLinkDTO.create(
+                    ResourceLinkTypeDTO.CHOOSE_UNIT,
+                    "Välj vårdenhet",
+                    "",
+                    true
+                )
             );
         }
+
         return resourceLinks;
     }
 
+    private boolean isChooseUnitAvailable(WebCertUser user) {
+        final var loggedInUnit =  user.getValdVardenhet();
+        return isOriginNormal(user.getOrigin()) && loggedInUnit != null;
```

Comment:
Isn't this the opposite logic of what should be implemented? If the logged in unit is not null then it means that the user has chosen a unit and we don't need to display the choose unit modal. Also, for change unit you included the origins NORMAL and UTHOPP. Here you only include NORMAL. Is this correct or should UTHOPP be included?

Reply:
This is correct, when you change unit both NORMAL and UTHOPP is included, but when you choose unit only NORMAL is included.